src/compiler/types.ts

```diff
@@ -7023,17 +7032,19 @@ namespace ts {
         TypeAssertions = 1 << 1,
         NonNullAssertions = 1 << 2,
         PartiallyEmittedExpressions = 1 << 3,
+        SatisfiesExpression = 1 << 4,

```

Comment:
I wonder if this needs to be different in any way than an `TypeAssertions`. I'm kind of surprised that we expose this enum publicly.